### PR TITLE
(CDAP-12987) Refactor the ArtifactManager to allow usage in unit test

### PR DESCRIPTION
**Pull request by chtyim**
- Have a local and remote implementation
- Added unit-test

### CODE REVIEW

**Review by albertshau — APPROVED** [review]
lgtm

**Comment by chtyim** [comment]
Build https://builds.cask.co/browse/CDAP-DUT6246

**Comment by chtyim** [comment]
Build passed. Merging.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/RemoteArtifactManager.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+                        @Assisted NamespaceId namespaceId, @Assisted RetryStrategy retryStrategy) {
+    super(cConf);
+    this.locationFactory = locationFactory;
+    this.authenticationContext = authenticationContext;
+    this.namespaceId = namespaceId;
+    this.retryStrategy = retryStrategy;
+    this.authorizationEnabled = cConf.getBoolean(Constants.Security.Authorization.ENABLED);
+    this.remoteClient = new RemoteClient(discoveryServiceClient, Constants.Service.APP_FABRIC_HTTP,
+                                         new DefaultHttpRequestConfig(false),
+                                         String.format("%s", Constants.Gateway.API_VERSION_3));
```
Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/RemoteArtifactManager.java:80]: No need to use `String.format` here, right?

**Pull request merged**